### PR TITLE
Coverage lines

### PR DESCRIPTION
**Pull request by rchoquet**
New coverage lines analyzer (and a small refactoring of request_date / is_internal_call / region_id)

~~Waiting for https://github.com/CanalTP/spark-stat-analyzer/pull/31 to be merged to create the migration~~

### CODE REVIEW

File: analyzers/coverage_journeys_requests_params.py [hunk 1]
```diff
@@ -14,9 +14,9 @@ def get_tuples_from_stat_dict(stat_dict):
             return []
 
         result.append((
-            datetime.utcfromtimestamp(stat_dict['request_date']).date(),
-            stat_dict['coverages'][0]['region_id'],
-            1 if 'canaltp' in stat_dict['user_name'] else 0  # is_internal_call
+            request_date(stat_dict),
+            region_id(stat_dict),
+            is_internal_call(stat_dict)
```
Comment (azime) [inline analyzers/coverage_journeys_requests_params.py]: :+1:

File: analyzers/coverage_lines.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
+from analyzers import Analyzer
+from analyzers.stat_utils import region_id, is_internal_call, request_date
+
+
+class AnalyzeCoverageLines(Analyzer):
+    @staticmethod
+    def get_lines(stat_dict):
+        if 'journeys' not in stat_dict or not len(stat_dict['journeys']):
+            return
+        for journey in stat_dict['journeys']:
```
Comment (azime) [inline analyzers/coverage_lines.py]:
````python
journeys=stat_dict.get('journeys', [])
if not len(journeys):

````

File: analyzers/coverage_lines.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            for section in journey_sections:
+                if not section.get('line_id'):
+                    continue
+
+                line_dict = {
+                    'type': section.get('type'),
+                    'line_id': section.get('line_id'),
+                    'line_code': section.get('line_code'),
+                    'network_id': section.get('network_id'),
+                    'network_name': section.get('network_name'),
```
Comment (azime) [inline analyzers/coverage_lines.py]:
````python
'line_code': section.get('line_code', ''),
'network_id': section.get('network_id', ''),
'network_name': section.get('network_name', ''),
````

File: analyzers/stat_utils.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+from datetime import datetime
+
+
+def region_id(stat_dict):
+    return stat_dict['coverages'][0]['region_id']
```
Comment (azime) [inline analyzers/stat_utils.py]: if len(stat_dict['coverages']) == 0

File: tests/stat_utils_test.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+def test_is_internal_call():
+    dict = {'user_name': 'foo-canaltp-bar'}
+    assert is_internal_call(dict) == 1
+    dict = {'user_name': 'foo-bar'}
+    assert is_internal_call(dict) == 0
+
+
+def test_request_date():
+    dict = {'request_date': 1484993662}
+    assert request_date(dict) == date(2017, 1, 21)
```
Comment (azime) [inline tests/stat_utils_test.py]: :+1:

File: migrations/alembic/versions/48218f5ebd1c_coverage_lines_table.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+"""
+
+# revision identifiers, used by Alembic.
+revision = '48218f5ebd1c'
+down_revision = '1cc79244cdf'
+
+from alembic import op
+from migrations.utils import get_create_partition_sql_func, get_drop_partition_sql_func
+import sqlalchemy as sa
+import geoalchemy2 as ga
```
Comment (azime) [inline migrations/alembic/versions/48218f5ebd1c_coverage_lines_table.py]: remove this line

**Comment by rchoquet** [comment]
New request, replacing 2.2.4 (<1s):

```sql
SELECT
    line_code,
    network_name,
    nb as nb_requests,
    row_number() over (order by nb DESC) as rang
FROM
    stat_compiled.coverage_lines cl
WHERE cl.type = 'public_transport'
  AND cl.line_code != ''
  AND cl.request_date >= :start_date
  AND cl.request_date < (:end_date :: date) + interval '1 day'
  AND cl.region_id = :coverage
  AND cl.is_internal_call = 0
ORDER BY nb_requests DESC
LIMIT 10;
```

**Comment by rchoquet** [comment]
Jenkins, test this please

**Pull request merged**